File: services/backend/internal/models/ll2-responses.go
package models

type LL2LaunchList struct {
	Count    int               `json:"count"`
	Launches []LL2LaunchNormal `json:"launches"`
}

type LL2AgencyList struct {
	Count    int                 `json:"count"`
	Agencies []LL2AgencyDetailed `json:"agencies"`
}

type LL2LauncherList struct {
	Count     int                       `json:"count"`
	Launchers []LL2LauncherConfigNormal `json:"launchers"`
}

type LL2LauncherFamilyList struct {
	Count    int                               `json:"count"`
	Families []LL2LauncherConfigFamilyDetailed `json:"families"`
}

type LL2LocationList struct {
	Count     int                             `json:"count"`
	Locations []LL2LocationSerializerWithPads `json:"locations"`
}

type LL2PadList struct {
	Count int      `json:"count"`
	Pads  []LL2Pad `json:"pads"`
}

File: apps/admin/src/types/ll2-responses.ts
import type { LL2LaunchNormal } from './ll2-launch';
import type { LL2AgencyDetailed } from './ll2-agency';
import type { LL2LauncherConfigNormal, LL2LauncherConfigFamilyDetailed } from './ll2-launcher';
import type { LL2LocationSerializerWithPads, LL2Pad } from './ll2-location';

export interface LL2LaunchList {
  count: number;
  launches: LL2LaunchNormal[];
}

export interface LL2AgencyList {
  count: number;
  agencies: LL2AgencyDetailed[];
}

export interface LL2LauncherList {
  count: number;
  launchers: LL2LauncherConfigNormal[];
}

export interface LL2LauncherFamilyList {
  count: number;
  families: LL2LauncherConfigFamilyDetailed[];
}

export interface LL2LocationList {
  count: number;
  locations: LL2LocationSerializerWithPads[];
}

export interface LL2PadList {
  count: number;
  pads: LL2Pad[];
}
